lib/config.js

```diff
@@ -18,6 +18,7 @@ function initConfig(startingConfig) {
   config['snapshotIntervalMs'] = 60 * 60 * 1000;
   config['beaconUrl'] = 'https://homebase.snyk.io/api/v1/beacon';
   config['snapshotUrl'] = `https://homebase.snyk.io/api/v2/snapshot/${startingConfig.projectId}/node`;
+  config['allowUnknownCA'] = false;
```

Comment:
the rest of our boolean flags are defaulted to true.. consider doing the opposite here, changing the flag to `restrictKnownCA` for example and set it to true by default

Reply:
hmm, interesting point.
this is my view:
1. the default value of other boolean flags is irrelevant, because their default value makes sense for their context. I don't think we want to impose this from unrelated context.

this already answers your comment, but it got me thinking a bit (which is usually not a good thing).

2. this flag is something that is being "opted in consciously". that is, we turn *on* a switch, which in my mind translates to "changing `false` to `true` in order to `allow` something that is usually blocked (or restricted)".

3. the next thing to consider, is how do we want to reflect this behaviour to our users? should they allow something that is usually blocked, or un-block something that usually blocked? I think the former is a bit more suitable.

4. the last thing I considered, is our usage inside the code, where we always negate its value: `!config['allowUnknownCA']` - wouldn't it be simpler to reverse the property's meaning and get rid of these exclamation marks?
I would argue that in this case, no. it might _look_ a bit simpler, but the logic of connecting the config value to `needle`'s `rejectUnauthorized` flag will still exist - it will just be a bit more implicit. which isn't _necessarily_ good.

what do you think?

Reply:
I think that psychologically it looks better to have a `restrictKnownCA`, it looks more secure. So we restrict it by default, and it is understandable that you turn some security measure off by turning this flag off. I don't see it as turning on a switch, but the other way around - we turn a security mechanism off

Reply:
`restrictKnownCA` makes me wonder about things like "wait, if I mention it now, what has it been doing so far?!" or "then what's the default value? have I been unprotected so far?"
which is kind of what crossed my mind when I looked at `Needle`'s `rejectUnauthorized` flag.
`allow...` leaves me with a feeling that, before mentioning it, I expect it to be disallowed, so everything is 🆒

Reply:
sounds good 👍 I still believe the second choice is better but don't have strong feelings about it, you can merge if you want :)